toolkits/ob-bi/carbon-home/deployment/siddhi-files/ManuallyAggregateMetricsApp.siddhi

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+@App:name("ManuallyAggregateMetricsApp")
+
+-- Stream to trigger metrics data aggregations
+@source(type='http', receiver.url='http://<BI_HOSTNAME>:8290/repopulate-cds-metrics', @map(type='json'))
+define stream TriggerRepopulateStream(fromTimestamp long, toTimestamp long);
+
+-- Raw metrics aggregation output stream
+@sink(type='inMemory',topic='apiInvocationRawData', @map(type='passThrough'))
+define stream CDSMetricsRawStream(id string, statusCode int, respTime long, electedResource string, reqTimestamp long,
+                        customerStatus string);
+
+-- Intermediate stream used for joining API_INVOCATION_RAW_DATA and API_LATENCY_RAW_DATA tables
+define stream FilteredInvocationsForJoin(messageId string, statusCode int, electedResource string, reqTimestamp long, customerStatus string);
+
+
+-- Stream to log each published metric event
+@sink(type='file',
+      file.uri='file:<LOG_PATH>/metrics-aggregate-events.log',
+      append='true',
+      @map(type='text', @payload("[INFO] [{{logTs}}] Publishing event with Id: {{id}}, EventTimestamp: {{reqTimestamp}}")))
+define stream CDSMetricsLoggingStream(logTs string, id string, reqTimestamp long);
+
+-- Stream to log the total count before aggregation begins
+@sink(type='file',
+      file.uri='file:<LOG_PATH>/metrics-aggregate-events.log',
```

Comment:
Who configures LOG_PATH ?

Reply:
We have given instructions in the readme to configure this. metrics-aggregate-events.log will be saved at this path.